pkg/bloomcompactor/config.go

```diff
@@ -14,13 +15,29 @@ type Config struct {
 	// section and the ingester configuration by default).
 	Ring ring.RingConfig `yaml:"ring,omitempty" doc:"description=Defines the ring to be used by the bloom-compactor servers. In case this isn't configured, this block supports inheriting configuration from the common ring section."`
 	// Enabled configures whether bloom-compactors should be used to compact index values into bloomfilters
-	Enabled           bool          `yaml:"enabled"`
-	WorkingDirectory  string        `yaml:"working_directory"`
-	MaxLookBackPeriod time.Duration `yaml:"max_look_back_period"`
+	Enabled            bool          `yaml:"enabled"`
+	WorkingDirectory   string        `yaml:"working_directory"`
+	MaxLookBackPeriod  time.Duration `yaml:"max_look_back_period"`
+	CompactionInterval time.Duration `yaml:"compaction_interval"`
+
+	// No need to add options to customize the retry backoff,
+	// given the defaults should be fine, but allow to override
+	// it in tests.
+	retryMinBackoff          time.Duration `yaml:"-"`
+	retryMaxBackoff          time.Duration `yaml:"-"`
+	CompactionRetries        int           `yaml:"compaction_retries" category:"advanced"`
+	TablesToCompact          int           `yaml:"tables_to_compact"`
```

Comment:
I don't think this should be a config. Say `tables_to_compact=4`, but we accept `7d` old data. That creates an easy misconfiguration that would never compact the older or newer tables, depending on which are preferred since each cycle would only take `4` from either end. Instead, I think we should have tenant configurations that specify the min and maximum durations we'll attempt compaction for (e.g. don't compact tables older than 7d or newer than 1d). This is similar to what I mentioned in a previous comment.